npm and deprecated `git://` URLs

The last few test runs failed on this

```
    npm ERR! Error while executing:
    npm ERR! /nix/var/nix/profiles/bknix-max/bin/git ls-remote -h -t git://github.com/SevInf/node-byline.git
    npm ERR! 
    npm ERR! fatal: remote error: 
    npm ERR!   The unauthenticated git protocol on port 9418 is no longer supported.
    npm ERR! Please see https://github.blog/2021-09-01-improving-git-protocol-security-github/ for more information.
    npm ERR! 
    npm ERR! exited with error code: 128
```

Their blog has more info at https://github.blog/2021-09-01-improving-git-protocol-security-github/ 

TLDR:

* Github is killing support for `git://` URLs
* Today is a "brownout" -- they disabled `git://` temporarily to cause things to break. Final removal is in March.
* Something in `npm` depends on a deprecated `git://` URL, so it fails.

--- a/bin/setup.sh
+++ b/bin/setup.sh
@@ -1,8 +1,6 @@
 #!/usr/bin/env bash
 set -e
 
-DEFAULT_MOSAICO_BRANCH="v0.15-civicrm-2"
-DEFAULT_MOSAICO_REPO="https://github.com/civicrm/mosaico"
 EXTROOT=$(cd `dirname $0`/..; pwd)
 EXTKEY="uk.co.vedaconsulting.mosaico"
 XMLBUILD="$EXTROOT/build/xml/schema"
@@ -83,24 +81,6 @@ function do_gencode() {
 
 ##############################
 function do_download() {
-  if [ ! -d "$EXTROOT/packages" ]; then
-    mkdir "$EXTROOT/packages"
-  fi
-  if [ ! -d "$EXTROOT/packages/mosaico" ]; then
-    git clone -b "$DEFAULT_MOSAICO_BRANCH" "$DEFAULT_MOSAICO_REPO" "$EXTROOT/packages/mosaico"
-  fi
-  pushd "$EXTROOT/packages/mosaico" >> /dev/null
-    local currentBranch=$(basename /$(git symbolic-ref HEAD 2>/dev/null))
-    if [ "$currentBranch" != "$DEFAULT_MOSAICO_BRANCH" ]; then
-      echo "Error: packages/mosaico is not on expected branch ($DEFAULT_MOSAICO_BRANCH). You may either:"
-      echo " (1) Checkout the branch '$DEFAULT_MOSAICO_BRANCH'. Then run 'setup.sh -D' again."
-      echo " (2) Manage the branch manualy. Be sure to call 'npm install' and 'grunt' as needed."
-      exit 1
-    fi
-    npm install
-    find node_modules -name '*.info' -delete
-    grunt build
-  popd >> /dev/null
   pushd "$EXTROOT" >> /dev/null
     composer install
   popd >> /dev/null

--- a/composer.json
+++ b/composer.json
@@ -1,9 +1,25 @@
 {
+  "name": "civipkg/uk.co.vedaconsulting.mosaico",
   "require": {
-    "intervention/image": "^2.4"
+    "intervention/image": "^2.4",
+    "civicrm/composer-downloads-plugin": "^3.0"
   },
   "replace": {
     "guzzlehttp/guzzle": "^6.5",
     "guzzlehttp/psr7": "^1.8.5"
+  },
+  "config": {
+    "allow-plugins": {
+      "civicrm/composer-downloads-plugin": true
+    }
+  },
+  "extra": {
+    "downloads": {
+      "mosaico": {
+        "version": "v0.15-civicrm-2.1",
+        "path": "packages/mosaico",
+        "url": "https://github.com/civicrm/mosaico/releases/download/{$version}/mosaico-dist.tar.gz"
+      }
+    }
   }
 }

